services/app-api/handlers/rate/shared/calculateAndPutRate.ts

```diff
@@ -19,6 +19,17 @@ export const formatMeasureData = (data: (Types.Measure | undefined)[]) => {
   });
 };
 
+//checks to see which set of data is valid, and only return those for calculation
+export const validateMeasureData = (data: FormattedMeasureData[]) => {
+  const validData: FormattedMeasureData[] = [];
+  data.forEach((program) => {
+    if (Object.values(program.rates).length > 0) {
+      validData.push(program);
+    }
+  });
+  return validData;
+};
+
```

Comment:
Suggestion: This is the classic use case for Array..filter:

```suggestion
// Checks to see which set of data is valid, and only return those for calculation
export const removeEmptyRates = (data: FormattedMeasureData[]) => {
  return data.filter((program) => Object.values(program.rates).length > 0);
};

```

Reply:
all hail the almighty filter! great catch, thanks!